components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java

```diff
@@ -1716,6 +1726,12 @@ Map loadAnnotations(SecurityContext ctx, Class nodeType, List nodeIDs,
 		}
 		try {
 		    IMetadataPrx service = gw.getMetadataService(ctx);
+            if (types == null || types.isEmpty())
+                dsFactory.getLogger().warn(
+                        this,
+                        "Loading *all* annotations for " + nodeType + " "
+                                + String.join(",", nodeIDs)
```

Comment:
This is a Java-8-ism. Use `com.google.common.base.Joiner.on(',').join(nodeIDs)`.